osu.Game/Overlays/Settings/Sections/Maintenance/DirectorySelectScreen.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using System.IO;
+using osu.Framework.Allocation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Screens;
+using osuTK;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Framework.Screens;
+
+namespace osu.Game.Overlays.Settings.Sections.Maintenance
+{
+    public abstract class DirectorySelectScreen : OsuScreen
+    {
+        private TriangleButton selectionButton;
+
+        private DirectorySelector directorySelector;
+
+        protected override OverlayActivation InitialOverlayActivationMode => OverlayActivation.Disabled;
+
+        protected abstract OsuSpriteText CreateHeader();
+
+        /// <summary>
+        /// Called upon selection of a directory by the user.
+        /// </summary>
+        /// <param name="directory">The selected directory</param>
+        protected abstract void OnSelection(DirectoryInfo directory);
+
+        protected virtual bool IsValidDirectory(DirectoryInfo info) => true;
+
+        protected virtual DirectoryInfo InitialPath => null;
+
+        public override bool AllowExternalScreenChange => false;
+
+        public override bool DisallowExternalBeatmapRulesetChanges => true;
```

Comment:
Not super sure about these being up here. These seem more suited to the migration screen more specifically, since they were used there to prevent the user from messing with things while the migration is taking place. While in general a directory selection screen doesn't really need to do any of this.